src/unix/mod.rs

```diff
@@ -726,7 +730,9 @@ extern {
 // TODO: get rid of this cfg(not(...))
 #[cfg(not(target_os = "android"))] // " if " -- appease style checker
 extern {
+    #[cfg(not(target_os = "haiku"))]
     pub fn getifaddrs(ifap: *mut *mut ifaddrs) -> ::c_int;
+    #[cfg(not(target_os = "haiku"))]
```

Comment:
Can these functions get pushed down into the lower modules instead of adding `#[cfg]` here?